packages/providers/mgt-msal2-provider/src/Msal2Provider.ts

```diff
@@ -0,0 +1,463 @@
+import { IProvider, LoginType, ProviderState, createFromProvider, Providers } from '@microsoft/mgt-element';
+import {
+  Configuration,
+  PublicClientApplication,
+  SilentRequest,
+  PopupRequest,
+  RedirectRequest,
+  AuthenticationResult,
+  AccountInfo,
+  EndSessionRequest,
+  InteractionRequiredAuthError
+} from '@azure/msal-browser';
+import { AuthenticationProviderOptions } from '@microsoft/microsoft-graph-client/lib/es/IAuthenticationProviderOptions';
+import { isRegExp } from 'util';
+
+export interface Msal2Config {
+  clientId: string;
+  loginType?: LoginType;
+  scopes?: string[];
+  loginHint?: string;
+  sid?: string;
+  domain_hint?: string;
+  redirectUri?: string;
+  authority?: string;
+  multiAccountDisabled?: boolean;
+  options?: Configuration;
+}
+export class Msal2Provider extends IProvider {
+  private _publicClientApplication: PublicClientApplication;
+
+  /**
+   * Login type, Either Redirect or Popup
+   *
+   * @private
+   * @type {LoginType}
+   * @memberof Msal2Provider
+   */
+  private _loginType: LoginType;
+
+  /**
+   * Login hint, if provided
+   *
+   * @private
+   * @memberof Msal2Provider
+   */
+  private _loginHint;
+
+  /**
+   * Domain hint if provided
+   *
+   * @private
+   * @memberof Msal2Provider
+   */
+  private _domainHint;
+
+  /**
+   * Session ID, if provided
+   *
+   * @private
+   * @memberof Msal2Provider
+   */
+  private _sid;
+
+  /**
+   * Configuration settings for authentication
+   *
+   * @private
+   * @type {Configuration}
+   * @memberof Msal2Provider
+   */
+  private ms_config: Configuration;
+
+  /**
+   * Gets the PublicClientApplication Instance
+   *
+   * @private
+   * @type {PublicClientApplication}
+   * @memberof Msal2Provider
+   */
+  public get publicClientApplication() {
+    return this._publicClientApplication;
+  }
+
+  /**
+   * List of scopes
+   *
+   * @type {string[]}
+   * @memberof Msal2Provider
+   */
+  public scopes: string[];
+
+  /**
+   * Account object
+   *
+   * @private
+   * @type {*}
+   * @memberof Msal2Provider
+   */
+  private _account: any;
```

Comment:
Is this the active account? If so, I'd name it `_activeAccount`. It doesn't seem to be necessary - can it be removed?